docs/source/advancedcommands.rst

```diff
@@ -1,6 +1,117 @@
+    m.substitutions.update({t_day: ("sweep", [8, 12, 16])})
+    sol = m.solve(verbosity=0)
+    self.assertEqual(len(sol["cost"]), 3)
+    npt.assert_allclose(sol(t_day) + sol(t_night), 24)
+
+
+Evaluated Free Variables
+------------------------
+
+Some free variables may be evaluated from the values of other (non-evaluated) free variables
+after the optimization is performed. For example, if the efficiency :math:`\nu` of a motor is not a GP-compatible
+variable, but :math:`(1-\nu)` is a valid GP variable, then :math:`\nu` can be calculated after solving.
+These evaluated free variables can be represented by a ``Variable`` with ``evalfn`` metadata.
+Note that this variable should not be used in constructing your model!
+
+.. code-block:: python
+
+    # code from t_constraints.test_evalfn in tests/t_sub.py
+    x = Variable("x")
+    x2 = Variable("x^2", evalfn=lambda v: v[x]**2)
+    m = Model(x, [x >= 2])
+    m.unique_varkeys = set([x2.key])
+    sol = m.solve(verbosity=0)
+    self.assertAlmostEqual(sol(x2), sol(x)**2)
+
+
+For evaluated variables than can be used during a solution, see ``externalfn`` under :ref:`sgp`.
+
+
+.. _Sweeps:
+
+Sweeps
+======
+
+Sweeps are useful for analyzing tradeoff surfaces. A sweep “value” is an Iterable of numbers, e.g. ``[1, 2, 3]``. The simplest way to sweep a model is to call ``model.sweep({sweepvar: sweepvalues})``, which will return a solution array but not change the model's substitutions dictionary. If multiple ``sweepvars`` are given, the method will run them all as independent one-dimensional sweeps and return a list of one solution per sweep. The method ``model.autosweep({sweepvar: (start, end)}, tol=0.01)`` behaves very similarly, except that only the bounds of the sweep need be specified and the region in betwen will be swept to a maximum possible error of tol in the log of the cost. For details see `1D Autosweeps`_ below.
+
+
+Sweep Substitutions
+-------------------
+Alternatively, or to sweep in multiple dimensions simultaneously, Variables can swept with a substitution value takes the form ``('sweep', Iterable), (e.g. 'sweep', np.linspace(1e6, 1e7, 100))``. During variable declaration, giving an Iterable value for a Variable is assumed to be giving it a sweep value: for example, ``x = Variable("x", [1, 2, 3])`` will sweep ``x`` over three values.
+
+Solving Sweeps
+--------------
+A Model with sweeps will solve for all possible combinations: e.g., if there’s a variable ``x`` with value ``('sweep', [1, 3])`` and a variable ``y`` with value ``('sweep', [14, 17])`` then the gp will be solved four times, for :math:`(x,y)\in\left\{(1, 14),\ (1, 17),\ (3, 14),\ (3, 17)\right\}`. The returned solutions will be a one-dimensional array (or 2-D for vector variables), accessed in the usual way.
+Sweeping Vector Variables
+
+Vector variables may also be substituted for: ``y = VectorVariable(3, "y", value=('sweep' ,[[1, 2], [1, 2], [1, 2]])`` will sweep :math:`y\ \forall~y_i\in\left\{1,2\right\}`.
```

Comment:
oh I see, here's the explanation I was asking for above -- maybe this section should be integrated with `Sweep Substitutions`, because it doesn't seem to apply to the `Model.sweep` and `Model.autosweep` versions?